src/theory/strings/theory_strings_rewriter.cpp

```diff
@@ -1817,107 +1817,123 @@ Node TheoryStringsRewriter::rewriteContains( Node node ) {
 
 Node TheoryStringsRewriter::rewriteIndexof( Node node ) {
   Assert(node.getKind() == kind::STRING_STRIDOF);
-  std::vector< Node > children;
-  getConcat( node[0], children );
-  //std::vector< Node > children1;
-  //getConcat( node[1], children1 );  TODO
-  std::size_t start = 0;
-  std::size_t val2 = 0;
-  if( node[2].isConst() ){
+  NodeManager* nm = NodeManager::currentNM();
+
+  std::vector<Node> children0;
+  getConcat(node[0], children0);
+  if (children0[0].isConst() && node[1].isConst() && node[2].isConst())
+  {
+    CVC4::String t = node[1].getConst<String>();
+    if (t.size() == 0)
```

Comment:
I suppose I could move this one outside as well.

My initial intention was to split this function into an "evaluation" section (where all arguments are constant) and the rest, since the general cases (below the if-then block at 1824) will catch cases like this.